src/balanced_ternary_converter.cpp

```diff
@@ -28,25 +28,23 @@ std::vector<int> make_ternary(const std::vector<int>& values, int value) {
     std::vector<int> ternary(values.size(), 0);
 
     for (std::size_t i = 0; i < values.size(); ++i) {
-        for (uint8_t step = 0; step < 2; ++step) {
-            if (value >= values[i]) {
-                ++ternary[i];
-                value -= values[i];
-            }
-        }
+        ternary[i] += value / values[i];
+        value %= values[i];
     }
+    std::reverse(ternary.begin(), ternary.end());
 
     return ternary;
 }
 
 void convert_to_balanced(std::vector<int>& ternary) {
-    for (int i = ternary.size() - 1; i >= 0; --i) {
+    for (int i = 0; i < ternary.size(); ++i) {
+        if (ternary[i] > 1 && i + 1 >= ternary.size()) ternary.push_back(0);
```

Comment:
```suggestion
        if (ternary[i] > 1 && i + 1 >= ternary.size()) {
            ternary.push_back(0);
        }
```

I always prefer to have brackets, keeps things easy to read.